yb-voyager/cmd/endMigrationCommand.go

```diff
@@ -579,64 +600,73 @@ func stopVoyagerCommands(msr *metadb.MigrationStatusRecord, lockFiles []*lockfil
 	}
 }
 
-func getLockFileForCommand(lockFiles []*lockfile.Lockfile, cmdName string) *lockfile.Lockfile {
-	for _, lockFile := range lockFiles {
-		if lockFile.GetCmdName() == cmdName {
-			return lockFile
-		}
-	}
-	return nil
-}
-
 // stop the voyager command by sending the signal to the process(if alive)
 func stopVoyagerCommand(lockFile *lockfile.Lockfile, signal syscall.Signal) {
 	if lockFile == nil || !lockFile.IsPIDActive() {
 		return
 	}
 
 	ongoingCmd := lockFile.GetCmdName()
-	utils.PrintAndLog("stopping the ongoing %q command", ongoingCmd)
-
 	ongoingCmdPID, err := lockFile.GetCmdPID()
 	if err != nil {
 		utils.ErrExit("getting PID of ongoing voyager command %q: %v", ongoingCmd, err)
 	}
 
-	log.Infof("stopping ongoing voyager commands %q with PID=%d", ongoingCmd, ongoingCmdPID)
+	fmt.Printf("stopping the ongoing command: %s\n", ongoingCmd)
+	log.Infof("stopping the ongoing command: %q with PID=%d", ongoingCmd, ongoingCmdPID)
 	err = stopProcessWithPID(ongoingCmdPID, signal)
 	if err != nil {
 		log.Warnf("stopping ongoing voyager command %q with PID=%d: %v", ongoingCmd, ongoingCmdPID, err)
 	}
 }
 
-func getCommandNamesFromLockFiles(lockFiles []*lockfile.Lockfile) []string {
-	var cmds []string
-	for _, lockFile := range lockFiles {
-		cmds = append(cmds, lockFile.GetCmdName())
-	}
-	return cmds
-}
-
 // this function wait for process to exit after signalling it to stop
 func stopProcessWithPID(pid int, signal syscall.Signal) error {
 	process, _ := os.FindProcess(pid) // Always succeeds on Unix systems
-
+	log.Infof("sending signal=%s to process with PID=%d", signal.String(), pid)
 	err := process.Signal(signal)
 	if err != nil {
 		return fmt.Errorf("sending signal=%s signal to process with PID=%d: %w", signal.String(), pid, err)
 	}
 
+	waitForProcessToExit(process)
+	return nil
+}
+
+func waitForProcessToExit(process *os.Process) {
 	// Reference: https://mezhenskyi.dev/posts/go-linux-processes/
-	// Poll for 10 sec to make sure process is terminated
+	// Poll every 2 sec to make sure process is stopped
 	// here process.Signal(syscall.Signal(0)) will return error only if process is not running
-	for i := 0; i <= 10; i++ {
-		time.Sleep(time.Second * 1)
-		err = process.Signal(syscall.Signal(0))
+	for {
+		time.Sleep(time.Second * 2)
+		err := process.Signal(syscall.Signal(0))
 		if err != nil {
-			return nil
+			log.Infof("process with PID=%d is stopped", process.Pid)
+			return
 		}
 	}
-	return nil
+}
+
+func stopDataExportCommand(lockFile *lockfile.Lockfile) {
+	if lockFile == nil || !lockFile.IsPIDActive() {
+		return
+	}
+
+	metaDB.UpdateMigrationStatusRecord(func(record *metadb.MigrationStatusRecord) {
+		record.EndMigrationRequested = true
```

Comment:
Add a comment here stating that dbzm plugin detects this flag change and then gracefully exits which in turn causes the `export data` command to exit cleanly.